packages/editor-ui/src/views/VariablesView.vue

```diff
@@ -155,33 +197,37 @@ function displayName(resource: EnvironmentVariable) {
 
 <template>
 	<ResourcesListLayout
-		ref="layout"
+		ref="layoutRef"
 		resource-key="variables"
 		:disabled="!isFeatureEnabled"
 		:resources="allVariables"
 		:initialize="initialize"
 		:shareable="false"
 		:displayName="displayName"
+		:sortFns="sortFns"
 		:sortOptions="['nameAsc', 'nameDesc']"
 		:showFiltersDropdown="false"
 		type="datatable"
 		:type-props="{ columns: datatableColumns }"
+		@sort="resetNewVariablesList"
+		@click:add="addTemporaryVariable"
 	>
 		<template #add-button>
-			<n8n-tooltip placement="top" :disabled="isFeatureEnabled">
+			<n8n-tooltip placement="top" :disabled="canCreateVariables">
 				<div>
 					<n8n-button
 						size="large"
 						block
-						:disabled="!isFeatureEnabled"
+						:disabled="!canCreateVariables"
 						@click="addTemporaryVariable"
 						data-test-id="resources-list-add"
 					>
 						{{ $locale.baseText(`variables.add`) }}
 					</n8n-button>
 				</div>
 				<template #content>
-					{{ i18n.baseText('variables.add.unavailable') }}
+					<span v-if="!isFeatureEnabled">{{ i18n.baseText('variables.add.unavailable') }}</span>
```

Comment:
The tooltip only shows when the user can't create variables, in the following 2 cases:
A. The user cannot create variables because the feature is not enabled -> show sales message
B. The user cannot create variables because the user does not have permissions -> show that instance owner can create variables